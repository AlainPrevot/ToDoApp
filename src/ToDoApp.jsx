import { AppRouter } from "./router/AppRouter";
import "./ToDoApp/event/handleAddEventListener";

export const ToDoApp = () => {
  return (
    <>
      <AppRouter />
    </>
  );
};
