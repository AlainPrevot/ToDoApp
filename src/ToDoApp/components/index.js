export * from './ModeStyle.jsx'
export * from './ToDoAdd.jsx'
export * from './ToDoContentListAdd.jsx'
export * from './ToDoDate.jsx'
export * from './ToDoItem'
export * from './ToDoList.jsx'
export * from './ToDoNav.jsx'
export * from './ToDoOptions.jsx'
export * from './ToDoOptionsButton.jsx'
export * from './ToDoOptionsDropdown.jsx'
