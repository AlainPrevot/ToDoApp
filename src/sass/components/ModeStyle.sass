#mode
    visibility: hidden

.modeDark, .modeLight
    background-position: center
    background-repeat: no-repeat
    background-size: cover
    cursor: pointer
    height: 36px
    position: fixed
    right: 18px
    top: 14px
    width: 36px

.modeLight
    background-image: url('../../Icons/Sol.png')
    display: none

.modeDark
    background-image: url('../../Icons/Luna.png')

#mode:checked ~ .modeLight
    display: block

#mode:checked ~ .modeDark
    display: none
