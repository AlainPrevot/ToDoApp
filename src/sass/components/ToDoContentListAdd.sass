.ToDoContentListAdd
    width: 90%
    height: 70%
    position: relative

.toDoContentListAdd__div
    height: 90%
    overflow-y: scroll

.toDoContentListAdd__div::-webkit-scrollbar
    -webkit-appearance: none

.toDoContentListAdd__div::-webkit-scrollbar:vertical
    width: 4px

.toDoContentListAdd__div::-webkit-scrollbar:horizontal
    height: 2px

.toDoContentListAdd__div::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoContentListAdd__div::-webkit-scrollbar-track
    border-radius: 10px

.toDoContentListAdd__div::-webkit-scrollbar-button:increment, .toDoContentListAdd__div::-webkit-scrollbar-button
    display: none

@media only screen and (min-width: 560px)

    .toDoContentListAdd__div
        height: 80%
