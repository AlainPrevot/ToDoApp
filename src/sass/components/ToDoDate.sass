.toDoDate
    margin: 50px 0
    text-align: center

    h2
        color: white
        font-size: 2.8rem

    h6
        color: #6E9CB9
        font-size: 1.2rem
        font-weight: 300
        margin: -0.7rem
