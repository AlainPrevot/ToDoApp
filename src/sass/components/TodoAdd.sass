
.toDoAdd__input
    background-color: #151D27
    border-radius: 6px
    border: none
    color: white
    height: 46px
    margin-bottom: 20px
    outline: none
    padding-left: 38px
    width: 100%

.toDoAdd__input::placeholder
    color: #6E9CB9
