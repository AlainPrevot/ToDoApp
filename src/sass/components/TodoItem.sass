.toDoItem
    cursor: pointer
    font-size: 1.8rem
    margin-left: 8px

.toDoItem--realizado
    color: var(--Color__primary)

    .toDoItem__span
        text-decoration: line-through

.toDoItem__span
    align-items: center
    display: flex
    font-size: 1rem
    height: 100%
    position: relative
    text-transform: capitalize
    top: -3px
