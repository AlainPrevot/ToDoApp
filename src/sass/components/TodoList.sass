.toDoList
    height: 400px
    overflow-x: hidden
    overflow-y: auto

.toDoList::-webkit-scrollbar
    -webkit-appearance: none

.toDoList::-webkit-scrollbar:vertical
    width: 4px

.toDoList::-webkit-scrollbar:horizontal
    height: 2px

.toDoList::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoList::-webkit-scrollbar-track
    border-radius: 10px

.toDoList::-webkit-scrollbar-button:increment, .toDoList::-webkit-scrollbar-button
    display: none
