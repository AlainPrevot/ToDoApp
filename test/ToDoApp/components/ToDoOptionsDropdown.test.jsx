import { render } from "@testing-library/react";
import { ToDoOptionsDropdown } from "../../../src/ToDoApp/components";

/* eslint-disable no-undef */
describe("Pruebas en <ToDoOptionsDropdown/>", () => {
  const ElementId = 1;
  test("Debe de hacer match con el Snapshot", () => {
    const { container } = render(<ToDoOptionsDropdown ElementId={ElementId} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de dibujarse correctamente", () => {
    const { container } = render(<ToDoOptionsDropdown ElementId={ElementId} />);

    expect(container.querySelector(".ToDoOptionsDropdown__box")).toBeTruthy();

    expect(
      container.querySelectorAll(".ToDoOptionsDropdown__box--icons").length
    ).toBe(3);
  });
});
