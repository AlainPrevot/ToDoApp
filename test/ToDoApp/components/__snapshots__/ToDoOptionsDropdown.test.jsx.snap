// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ToDoOptionsDropdown/> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="ToDoOptionsDropdown__box menu"
  >
    <div
      class="ToDoOptionsDropdown__box--icons menu"
    />
    <div
      class="ToDoOptionsDropdown__box--icons menu"
    />
    <div
      class="ToDoOptionsDropdown__box--icons menu"
    />
  </div>
</div>
`;
